src/wazuh_modules/wm_gcp.c

```diff
@@ -312,80 +321,67 @@ void wm_gcp_pubsub_run(const wm_gcp_pubsub *data) {
-                mtwarn(WM_GCP_PUBSUB_LOGTAG, "Unknown error.");
-            else
-                mtwarn(WM_GCP_PUBSUB_LOGTAG, "%s", unknown_error_msg);
-        } else if(status == 2) {
-            char * ptr;
-            if (ptr = strstr(output, "integration.py: error:"), ptr) {
-                ptr += 16;
-                mtwarn(WM_GCP_PUBSUB_LOGTAG, "Error parsing arguments: %s", ptr);
-            } else {
-                mtwarn(WM_GCP_PUBSUB_LOGTAG, "Error parsing arguments.");
-            }
-        } else {
-            char * ptr;
-            if (ptr = strstr(output, "ERROR: "), ptr) {
-                ptr += 7;
-                mtwarn(WM_GCP_PUBSUB_LOGTAG, "%s", ptr);
-            } else {
-                mtwarn(WM_GCP_PUBSUB_LOGTAG, "%s", output);
-            }
-        }
-        mtdebug1(WM_GCP_PUBSUB_LOGTAG, "OUTPUT: %s", output);
     }
 
+    wm_gcp_parse_output(output, WM_GCP_PUBSUB_LOGTAG);
+    os_free(output);
+}
+
+static void wm_gcp_parse_output(char *output, char *tag){
     char *line;
-    char *save_ptr = NULL;
-
-    for (line = strtok_r(output, "\n", &save_ptr); line; line = strtok_r(NULL, "\n", &save_ptr)) {
-        switch (data->logging) {
-            case 0:
-                mtinfo(WM_GCP_PUBSUB_LOGTAG, "Logging disabled.");
-                break;
-            case 1:
-                if (strstr(line, "- DEBUG -")) {
-                    mtdebug1(WM_GCP_PUBSUB_LOGTAG, "%s", line);
-                } else if (!strstr(line, "- WARNING -") && !strstr(line, "- INFO -")
-                && !strstr(line, "- ERROR -") && !strstr(line, "- CRITICAL -")) {
-                    mtdebug1(WM_GCP_PUBSUB_LOGTAG, "%s", line);
-                }
-                break;
-            case 2:
-                if (line = strstr(line, "- INFO -"), line) {
-                    mtinfo(WM_GCP_PUBSUB_LOGTAG, "%s", line);
-                }
-                break;
-            case 3:
-                if (line = strstr(line, "- WARNING -"), line) {
-                    mtwarn(WM_GCP_PUBSUB_LOGTAG, "%s", line);
-                }
-                break;
-            case 4:
-                if (line = strstr(line, "- ERROR -"), line) {
-                    mterror(WM_GCP_PUBSUB_LOGTAG, "%s", line);
-                }
-                break;
-            case 5:
-                if (line = strstr(line, "- CRITICAL -"), line) {
-                    mterror(WM_GCP_PUBSUB_LOGTAG, "%s", line);
-                }
-                break;
-            default:
-                if (line = strstr(line, "- INFO -"), line) {
-                    mtinfo(WM_GCP_PUBSUB_LOGTAG, "%s", line);
-                }
-                break;
+    char * parsing_output = output;
+    int debug_level = isDebug();
+
+    for (line = strstr(parsing_output, WM_GCP_LOGGING_TOKEN); line; line = strstr(parsing_output, WM_GCP_LOGGING_TOKEN)) {
+        char * tokenized_line;
+        os_calloc(WM_STRING_MAX, sizeof(char), tokenized_line);
+	char * next_lines;
```

Comment:
Please, fix line indentation.

Reply:
Sorry, my code editor mixed tabs and spaces :angry: